### PR TITLE
update

### PR DESCRIPTION
**Pull request by liangguodong00100**
trigger add  popupContainerId props

### CODE REVIEW

File: src/widgets/trigger/index.js [hunk 1]
```diff
@@ -142,13 +142,15 @@ class Trigger extends React.Component<TriggerProps, TriggerState> {
...
   popupContainer: ?Object;
 
-  getContainer() {
-    if (this.popupContainer) {
+  getContainer(popupContainerId?: string) {
+    if (popupContainerId && this.popupContainer) {
+      return this.popupContainer;
+    } else if (this.popupContainer) {
       return this.popupContainer;
     }
```
Comment (SleepyAries) [inline src/widgets/trigger/index.js:148]: 判断条件重复

Comment (hadesXx) [reply src/widgets/trigger/index.js:148]: componentDidMount已经判断过 this.popupContainer 的取值了，直接return this.popupContainer 就好了

File: src/widgets/trigger/index.js [hunk 2]
```diff
@@ -209,6 +211,13 @@ class Trigger extends React.Component<TriggerProps, TriggerState> {
     });
   }
 
+  componentDidMount() {
+    const { popupContainerId } = this.props;
+    if (popupContainerId) {
+      this.popupContainer = document.getElementById(popupContainerId);
```
Comment (hadesXx) [inline src/widgets/trigger/index.js]: 如果 popupContainerId 存在，也要判断 document.getElementById是否有值，如果没有值，就不应该 将this.popupContainer 赋值

File: src/widgets/trigger/index.js [hunk 3]
```diff
@@ -209,6 +209,14 @@ class Trigger extends React.Component<TriggerProps, TriggerState> {
...
   }
 
+  componentDidMount() {
+    const { popupContainerId } = this.props;
+    const geTriggerContainerDom = document.getElementById(popupContainerId);
+    if (popupContainerId && geTriggerContainerDom) {
+      this.popupContainer = geTriggerContainerDom;
+    }
+  }
+
```
Comment (hadesXx) [inline src/widgets/trigger/index.js:222]: componentDidMount() {
        const { popupContainerId } = this.props;
        const geTriggerContainerDom = document.getElementById(popupContainerId);
        if (popupContainerId && geTriggerContainerDom) {
            const targetPosition = window.getComputedStyle(geTriggerContainerDom).position;
            geTriggerContainerDom.style.position = targetPosition === 'static' ? 'relative' : targetPosition;
            this.popupContainer = geTriggerContainerDom;
        }
    }
 默认加一个 relative 比较好。代码写好了

File: src/widgets/trigger/index.js [hunk 4]
```diff
@@ -209,6 +209,17 @@ class Trigger extends React.Component<TriggerProps, TriggerState> {
     });
   }
 
+  componentDidMount() {
+    const { popupContainerId } = this.props;
+    const getTriggerContainerDom = document.getElementById(popupContainerId);
+    if (popupContainerId && getTriggerContainerDom) {
```
Comment (zenjava) [inline src/widgets/trigger/index.js]: triggerContainerDom  = document.getElementById(popupContainerId);

if(triggerContainerDom)

**Review by zenjava — CHANGES_REQUESTED** [review]

**Review by zenjava — APPROVED** [review]

**Pull request merged**